plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/StandardDeployPreferencesPanel.java

```diff
@@ -141,33 +144,29 @@ private void setupDataBinding() {
   }
 
   private void setupAccountEmailDataBinding(DataBindingContext context) {
-    IValidator accountSelectedChecker = new IValidator() {
-      @Override
-      public IStatus validate(Object value /* email */) {
-        if (requireValues && Strings.isNullOrEmpty((String) value)) {
-          if (accountSelector.isSignedIn()) {
-            return ValidationStatus.error(Messages.getString("error.account.missing.signedin"));
-          } else {
-            return ValidationStatus.error(Messages.getString("error.account.missing.signedout"));
+    AccountSelectorObservableValue accountSelectorObservableValue =
+        new AccountSelectorObservableValue(accountSelector);
+    UpdateValueStrategy modelToTarget =
+        new UpdateValueStrategy().setConverter(new Converter(String.class, String.class) {
+          @Override
+          public Object convert(Object fromObject /* email */) {
+            return accountSelector.isEmailAvailable((String) fromObject) ? fromObject : null;
```

Comment:
can you rewrite this without the ?: operator. I had to read this three times before I saw it was a simple not null check. Oh hell, that isn't it either, is it? Simple if-else would make this much easier to read, at least for me.